embassy-stm32/src/usart/mod.rs

```diff
@@ -211,6 +211,30 @@ impl Default for Config {
     }
 }
 
+#[derive(Clone, Copy, PartialEq, Eq, Debug)]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+/// Half duplex IO mode
+pub enum HalfDuplexConfig {
+    /// Push pull allows for faster baudrates, may require series resistor
+    PushPull,
+    /// Open drain output using external pull up resistor
+    OpenDrainExternal,
+    #[cfg(not(gpio_v1))]
+    /// Open drain output using internal pull up resistor
+    OpenDrainInternal,
+}
+
+impl HalfDuplexConfig {
+    pub fn af_type(self) -> gpio::AfType {
```

Comment:
i'd make this non-pub, there's no reason for the user to call it plus this way you don't need to doc it.

Reply:
Of course